lib/route.js

```diff
@@ -344,6 +345,15 @@ function buildRouting (options) {
       }
     }
 
+    const connHeader = String.prototype.toLowerCase.call(req.headers.connection || '')
+    if (connHeader === 'keep-alive') {
+      keepAliveConnections.add(req.socket)
+      req.socket.on('close', removeTrackedSocket.bind({ keepAliveConnections, socket: req.socket }))
```

Comment:
You are adding an event listener for every incoming request on that given socket. You should do this only for sockets that have never been added to the set.